### DISCUSSION TITLE
This lib will only support react?

### DISCUSSION THREAD

**Discussion by nhattran998** [kuma-ui/kuma-ui#137] [Q&A] [answered]
Hi @poteboy, thank you for the awesome lib! But in the future, this will extend more lib like Solid, Preact, Vue...?

**Comment by poteboy** [comment, marked as answer]
@harrytran998 
Yes, as long as it's written in JSX (I believe Vue also supports JSX syntax), I plan to support it. However, I'm currently busy with other developments (see: https://github.com/poteboy/kuma-ui/issues/127), so it's not a high priority at the moment.

**Reply by Jaeiya** [reply]
I'm also looking forward to SolidJS support. Once I realized that PandaCSS doesn't support Zero-runtime CSS, that was a deal breaker for me. This is the best of both worlds.

**Comment by kfranqueiro** [comment]
A friend informed me of the existence of this library last week and I'm pretty excited about it, and would definitely love to see it eventually broaden support to other libraries.

FYI I briefly tried throwing it at preact, and components like `Box` and `Button` seem to work but `k` gives an error about not being usable at runtime. (I didn't go as far as tinkering with theme stuff.)

I also tried it with Qwik but it didn't seem to get off the ground, with console warnings about duplicate implementations of "JSXNode" found.

**Comment by poteboy** [comment]
@Jaeiya @kfranqueiro 
I have discussed with the other maintainers, and we have decided to potentially support @astrojs/react, Preact, and Solid.js, in that order of priority, if feasible. As for other frameworks not mentioned above, we do not have plans to support them at the moment.
